src/app/app.component.ts

```diff
@@ -1,14 +1,27 @@
 import { Component, OnInit } from '@angular/core';
 import { AuthService } from './services/auth.service';
+import { Router } from '@angular/router';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
 export class AppComponent implements OnInit {
-  constructor(private authService: AuthService) { }
+  constructor(private authService: AuthService, private router: Router) { }
   ngOnInit(): void {
-   this.authService.checkTokenValidity();
+    if (window.location.href.includes('resetPassword')) {
```

Comment:
why don't you use the separate component?

Reply:
Didn't solved, becouse we have hash and it's the only way to make redirect.